xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/XWiki.java

```diff
@@ -4067,19 +4067,11 @@ public void sendValidationEmail(String xwikiname, String password, String email,
 
         }
 
-        try {
-            VelocityContext vcontext = (VelocityContext) context.get("vcontext");
-            vcontext.put(addfieldname, addfieldvalue);
-            vcontext.put("email", email);
-            vcontext.put("password", password);
-            vcontext.put("sender", sender);
-            vcontext.put("xwikiname", xwikiname);
-            content = parseContent(content, context);
-        } catch (Exception e) {
-            throw new XWikiException(XWikiException.MODULE_XWIKI_EMAIL,
-                XWikiException.ERROR_XWIKI_EMAIL_CANNOT_PREPARE_VALIDATION_EMAIL,
-                "Exception while preparing the validation email", e, null);
-
+        if(content != null) {
+            content =
+                evaluateMailContent(xwikiname, password, email, addfieldname, addfieldvalue, context, sender, content);
+        } else {
+            // TODO: load from template, if template exists, otherwisejust return the empty string?
```

Comment:
Given the open API where any value can be set for `contentfield`, my current understand is that we need to check for the presence of a template with a corresponding name. We should maybe define a convention like a special `mail` sub-directory.